Sources/Edge.swift

```diff
@@ -231,4 +232,10 @@ public class Edge: NSObject, Extension {
         return ConsentStatus.getCollectConsentOrDefault(eventData: consentXDMSharedState)
     }
 
+    /// Returns current `ImplementationDetails` for this session.
+    /// - Returns: the `ImplementationDetails` for this session or nil if not yet set
+    private func getImplementationDetails() -> [String: Any]? {
+        return state?.implementationDetails
```

Comment:
since you are calling this from the hit processor, you might want to protect the access to it